6_3_iterable.js

```diff
@@ -0,0 +1,48 @@
+class Group {
+    constructor(array) {
+        this.array = array;
+    }
+
+    static from(array) {
+        let group = new Group(array);
+        return group;
+    }
+
+    add(value) {
+        this.array.includes(value) || this.array.push(value);
+        return value;
+    }
+
+    delete(value) {
+        this.array.includes(value) && this.array.splice(this.array.indexOf(value), 1);
+        return value;
+    }
+
+    has(value) {
+        return this.array.some(item => item === value);
+    }
+
+    [Symbol.iterator]() {
+        return new GroupIterator(this);
+    }
+}
+
+class GroupIterator {
+    constructor(cluster) {
+        (this.cluster = cluster), (this.position = 0);
+    }
+    next() {
+        if (this.position < this.cluster.array.length) {
+            let result = { done: false, value: this.cluster.array[this.position] };
+            this.position++;
```

Comment:
I think we should avoid post incrementation operator and we should do this explicitly `this.position = this.position + 1`. In bigger project it wouldn't be easy to read when u mutate some value with ++ operator.